plugin/trino-iceberg/src/test/java/io/trino/plugin/iceberg/BaseIcebergConnectorSmokeTest.java

```diff
@@ -870,6 +871,50 @@ public void testMetadataDeleteAfterCommitEnabled()
         assertUpdate("DROP TABLE " + tableName);
     }
 
+    @Test
+    public void testIcebergTablesSystemTable()
+            throws Exception
+    {
+        String schemaName = getSession().getSchema().orElseThrow();
+        String firstSchema = "first_schema_" + randomNameSuffix();
+        String secondSchema = "second_schema_" + randomNameSuffix();
+        String firstSchemaLocation = schemaPath().replaceAll(schemaName, firstSchema);
+        String secondSchemaLocation = schemaPath().replaceAll(schemaName, secondSchema);
+        assertQuerySucceeds("CREATE SCHEMA " + firstSchema + " WITH (location = '%s')".formatted(firstSchemaLocation));
+        assertQuerySucceeds("CREATE SCHEMA " + secondSchema + " WITH (location = '%s')".formatted(secondSchemaLocation));
+        QueryRunner queryRunner = getQueryRunner();
+        Session firstSchemaSession = Session.builder(queryRunner.getDefaultSession()).setSchema(firstSchema).build();
+        Session secondSchemaSession = Session.builder(queryRunner.getDefaultSession()).setSchema(secondSchema).build();
+
+        try (TestTable _ = new TestTable(
+                sql -> getQueryRunner().execute(firstSchemaSession, sql),
+                "first_schema_table1_",
+                "(id int)");
+                TestTable _ = new TestTable(
+                        sql -> getQueryRunner().execute(firstSchemaSession, sql),
+                        "first_schema_table2_",
+                        "(id int)");
+                TestTable secondSchemaTable = new TestTable(
+                        sql -> queryRunner.execute(secondSchemaSession, sql),
+                        "second_schema_table_",
+                        "(id int)");
+                AutoCloseable _ = createAdditionalTables(firstSchema)) {
+            assertThat(query("SELECT * FROM iceberg.system.iceberg_tables WHERE table_schema = '%s'".formatted(firstSchema)))
+                    .matches("SELECT table_schema, table_name FROM iceberg.information_schema.tables WHERE table_schema='%s'".formatted(firstSchema));
+            assertThat(query("SELECT * FROM iceberg.system.iceberg_tables WHERE table_schema in ('%s', '%s')".formatted(firstSchema, secondSchema)))
+                    .matches("SELECT table_schema, table_name FROM iceberg.information_schema.tables WHERE table_schema IN ('%s', '%s')".formatted(firstSchema, secondSchema));
+        }
+        finally {
+            assertQuerySucceeds("DROP SCHEMA " + firstSchema);
+            assertQuerySucceeds("DROP SCHEMA " + secondSchema);
+        }
+    }
+
+    protected AutoCloseable createAdditionalTables(String schema)
+    {
+        return () -> {};
+    }
+
```

Comment:
We could add another test to `BaseTrinoCatalogTest` - creates a Hive table and verify the table doesn't appear in `iceberg_tables` system table.